File: core/src/com/bombbird/terminalcontrol/entities/aircrafts/AircraftType.java
package com.bombbird.terminalcontrol.entities.aircrafts;

import com.bombbird.terminalcontrol.utilities.saving.FileLoader;

import java.util.HashMap;

public class AircraftType {
    private static final HashMap<String, int[]> aircraftTypes = FileLoader.loadAircraftData();

    public static char getWakeCat(String type) {
        if (!aircraftTypes.containsKey(type)) return 'M';
        return (char) aircraftTypes.get(type)[0];
    }

    public static char getRecat(String type) {
        if (!aircraftTypes.containsKey(type)) return 'D';
        return (char) aircraftTypes.get(type)[5];
    }

    public static int getV2(String type) {
        if (!aircraftTypes.containsKey(type)) return 150;
        return aircraftTypes.get(type)[1];
    }

    public static int getTypClimb(String type) {
        if (!aircraftTypes.containsKey(type)) return 2000;
        return aircraftTypes.get(type)[2];
    }

    public static int getTypDes(String type) {
        if (!aircraftTypes.containsKey(type)) return 2000;
        return aircraftTypes.get(type)[3];
    }

    public static int getApchSpd(String type) {
        if (!aircraftTypes.containsKey(type)) return 150;
        return aircraftTypes.get(type)[4];
    }
}

File: core/src/com/bombbird/terminalcontrol/entities/aircrafts/AircraftType.kt
package com.bombbird.terminalcontrol.entities.aircrafts

import com.bombbird.terminalcontrol.utilities.saving.FileLoader.loadAircraftData

object AircraftType {
    private val aircraftTypes = loadAircraftData()
    fun getWakeCat(type: String): Char {
        return if (!aircraftTypes.containsKey(type)) 'M' else aircraftTypes[type]!![0].toChar()
    }

    @JvmStatic
    fun getRecat(type: String): Char {
        return if (!aircraftTypes.containsKey(type)) 'D' else aircraftTypes[type]!![5].toChar()
    }

    fun getV2(type: String): Int {
        return if (!aircraftTypes.containsKey(type)) 150 else aircraftTypes[type]!![1]
    }

    fun getTypClimb(type: String): Int {
        return if (!aircraftTypes.containsKey(type)) 2000 else aircraftTypes[type]!![2]
    }

    fun getTypDes(type: String): Int {
        return if (!aircraftTypes.containsKey(type)) 2000 else aircraftTypes[type]!![3]
    }

    fun getApchSpd(type: String): Int {
        return if (!aircraftTypes.containsKey(type)) 150 else aircraftTypes[type]!![4]
    }

    fun getMaxCruiseSpd(type: String): Int {
        return if (!aircraftTypes.containsKey(type)) -1 else aircraftTypes[type]!![6]
    }

    fun getMaxCruiseAlt(type: String): Int {
        return if (!aircraftTypes.containsKey(type)) 43000 else aircraftTypes[type]!![7]
    }
}